packages/expo-modules-core/ios/FileSystemUtilities/FileSystemUtilities.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+
+public enum FileSystemPermissionFlags {
+  case none
+  case read
+  case write
+}
+
+public struct FileSystemUtilities {
+  private let appContext: AppContext?
+
+  public init(appContext: AppContext?) {
+    self.appContext = appContext
+  }
+
+  public func ensureDirExists(at url: URL?) -> Bool {
+    guard let url else {
+      return false
+    }
+    let exists = FileManager.default.fileExists(atPath: url.path)
+
+    if !exists {
+      do {
+        try FileManager.default.createDirectory(atPath: url.path, withIntermediateDirectories: true)
+      } catch {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  public func generatePath(in directory: URL?, with ext: String) -> String {
+    guard let directory else {
+      return ""
+    }
+    let fileName = UUID().uuidString.appending(ext)
+    _ = ensureDirExists(at: directory)
```

Comment:
Probably better to add `@discardableResult` to `ensureDirExists` so we don't have to do the assignment